### PR TITLE
feat(dashboard): Retrigger Deployments

### LINKED ISSUE #1457: Retrigger deployments
## Problem

Our infrastructure supports retrigger deployments, but there is no way of triggering a deployment in the dashboard.

## Solution

Add a Retrigger button to the latest non-in-progress deployment.

### PR DESCRIPTION
**Pull request by szilarddoro**
Resolves #1457. There is [a backend change](https://github.com/nhost/console-next/pull/1544) that adds the support for inserting deployments into the `deployments` table.

I unified the component we are using on the Overview and the Deployments page.

### CODE REVIEW

File: dashboard/src/components/applications/AppDeployments.tsx [hunk 1]
```diff
@@ -226,42 +79,61 @@ export default function AppDeployments(props: AppDeploymentsProps) {
...
-            {data.deployments.map((deployment) => (
-              <AppDeploymentRow
-                deployment={deployment}
+          <List className="mt-3 divide-y-1 border-t border-b">
+            {deployments.map((deployment, index) => (
+              <DeploymentListItem
                 key={deployment.id}
-                isDeploymentLive={idOfLiveDeployment === deployment.id}
+                deployment={deployment}
+                isLive={liveDeploymentId === deployment.id}
```
Comment (guicurcio) [inline dashboard/src/components/applications/AppDeployments.tsx:124]: Thanks for merging these components together. But should we even show the redeploy button here? I think it looks a bit too crowded:

File: dashboard/src/components/deployments/DeploymentListItem/DeploymentListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const { commitMessage } = deployment;
+
+  return (
+    <ListItem.Root>
+      <ListItem.Button
+        className="grid grid-flow-col items-center justify-between gap-2 px-2 py-2"
+        component={NavLink}
+        href={`/${currentWorkspace.slug}/${currentApplication.slug}/deployments/${deployment.id}`}
+      >
+        <div className="flex cursor-pointer flex-row items-center justify-center space-x-2 self-center">
```
Comment (guicurcio) [inline dashboard/src/components/deployments/DeploymentListItem/DeploymentListItem.tsx:69]: Can we add some variants for mobile?

Comment (guicurcio) [reply dashboard/src/components/deployments/DeploymentListItem/DeploymentListItem.tsx:69]: 2023-01-11 08_38_01-Nhost Netlify Starter Template - Nhost

I think that hiding the commit sha and the time should be good wdyt?

**Review by guicurcio — CHANGES_REQUESTED** [review]

**Review by guicurcio — APPROVED** [review]
Given that most of the required changes are actually design-based (and were not part of this project) I think we could merge this and then make those cosmetic changes :thinking:

File: dashboard/src/components/deployments/DeploymentListItem/DeploymentListItem.tsx (hunk 2 above)
Comment (guicurcio) [reply dashboard/src/components/deployments/DeploymentListItem/DeploymentListItem.tsx:69]: Szilard, I think we could convert both of these into issues and come back later when the designs are ready

**Comment by szilarddoro** [comment]
Created an issue for this design improvement: https://github.com/nhost/nhost/issues/1510

**Pull request merged**